src/common/tests_dmg_helpers.c

```diff
@@ -0,0 +1,503 @@
+{
+	int			argcount = 0;
+	char			**args = NULL;
+	struct passwd		*passwd = NULL;
+	struct group		*group = NULL;
+	daos_mgmt_pool_info_t	pool_info = {};
+	struct json_object	*dmg_out = NULL;
+	int			rc = 0;
+
+	if (grp != NULL) {
+		args = cmd_push_arg(args, &argcount,
+				    "--sys=%s ", grp);
+		if (args == NULL)
+			D_GOTO(out, rc = -DER_NOMEM);
+	}
+
+	if (tgts != NULL) {
+		char *ranks_str = rank_list_to_string(tgts);
+
+		if (ranks_str == NULL) {
+			D_ERROR("failed to create rank string\n");
+			D_GOTO(out_cmd, rc = -DER_NOMEM);
+		}
+		args = cmd_push_arg(args, &argcount,
+				    "--ranks=%s ", ranks_str);
+		D_FREE(ranks_str);
+		if (args == NULL)
+			D_GOTO(out, rc = -DER_NOMEM);
+	}
+
+	passwd = getpwuid(uid);
+	if (passwd == NULL) {
+		D_ERROR("unable to resolve %d to passwd entry\n", uid);
+		D_GOTO(out_cmd, rc = -DER_INVAL);
+	}
+
+	args = cmd_push_arg(args, &argcount,
+			    "--user=%s ", passwd->pw_name);
+	if (args == NULL)
+		D_GOTO(out, rc = -DER_NOMEM);
+
+	group = getgrgid(gid);
+	if (group == NULL) {
+		D_ERROR("unable to resolve %d to group name\n", gid);
+		D_GOTO(out_cmd, rc = -DER_INVAL);
+	}
+
+	args = cmd_push_arg(args, &argcount,
+			    "--group=%s ", group->gr_name);
+	if (args == NULL)
+		D_GOTO(out, rc = -DER_NOMEM);
+
+	args = cmd_push_arg(args, &argcount,
+			    "--scm-size=%"PRIu64"b ", scm_size);
+	if (args == NULL)
+		D_GOTO(out, rc = -DER_NOMEM);
+
+	if (nvme_size > 0) {
+		args = cmd_push_arg(args, &argcount,
+				    "--nvme-size=%"PRIu64"b ", nvme_size);
+		if (args == NULL)
+			D_GOTO(out, rc = -DER_NOMEM);
+	}
+
+	rc = daos_dmg_json_pipe("pool create", dmg_config_file,
+				args, argcount, &dmg_out);
+	if (rc != 0) {
+		D_ERROR("dmg failed");
+		goto out_json;
+	}
+
+	rc = parse_pool_info(dmg_out, &pool_info);
+	if (rc != 0) {
+		D_ERROR("failed to parse pool info\n");
+		goto out_json;
+	}
+
+	uuid_copy(uuid, pool_info.mgpi_uuid);
+	if (svc == NULL)
+		goto out_svc;
+
+	rc = d_rank_list_copy(svc, pool_info.mgpi_svc);
+	if (rc != 0) {
+		D_ERROR("failed to dup svc rank list\n");
+		goto out_svc;
+	}
+
+out_svc:
+	d_rank_list_free(pool_info.mgpi_svc);
+out_json:
+	if (dmg_out != NULL)
+		json_object_put(dmg_out);
+out_cmd:
+	cmd_free_args(args, argcount);
+out:
+	return rc;
+}
+
+int
+dmg_pool_destroy(const char *dmg_config_file,
+		 const uuid_t uuid, const char *grp, int force)
+{
+	char			uuid_str[DAOS_UUID_STR_SIZE];
+	int			argcount = 0;
+	char			**args = NULL;
+	struct json_object	*dmg_out = NULL;
+	int			rc = 0;
+
+	uuid_unparse_lower(uuid, uuid_str);
+	args = cmd_push_arg(args, &argcount,
+			    "--pool=%s ", uuid_str);
+	if (args == NULL)
+		D_GOTO(out, rc = -DER_NOMEM);
+
+	if (force != 0) {
+		args = cmd_push_arg(args, &argcount, "--force");
+		if (args == NULL)
+			D_GOTO(out, rc = -DER_NOMEM);
+	}
+
+	rc = daos_dmg_json_pipe("pool destroy", dmg_config_file,
+				args, argcount, &dmg_out);
+	if (rc != 0) {
+		D_ERROR("dmg failed");
+		goto out_json;
+	}
+
+out_json:
```

Comment:
Looks like no need for this "out_json" label for now since only used once and it's next door. Maybe need for future code expansion?

Reply:
Yeah, I figured I'd just leave it. There's precedent for this style in the dRPC handlers, and it reduces the chance for errors if new code is added after this.